ostd/src/io_mem.rs

```diff
@@ -61,10 +106,7 @@ impl IoMem {
         assert!(!range.is_empty() && range.end <= self.limit);
 
         // We've checked the range is in bounds, so we can construct the new `IoMem` safely.
-        Self {
-            virtual_address: self.virtual_address + range.start,
-            limit: range.end - range.start,
-        }
+        Self::new(range.start + self.paddr()..range.end + self.paddr())
```

Comment:
This is not quite good. It allocates a new area and do the mapping. You can let two `IoMem`s share the same area using `Arc`, which is better.